src/com/xilinx/rapidwright/util/RunVivado.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2021-2022, Xilinx, Inc.
+ * Copyright (c) 2022-2023, Advanced Micro Devices, Inc.
+ * All rights reserved.
+ *
+ * Author: Zak Nafziger, Xilinx Research Labs.
+ *
+ * This file is part of RapidWright.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.xilinx.rapidwright.util;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Utility methods to run a tcl script through vivado and parse the stdout
+ * @author zakn
+ *
+ */
+public class RunVivado {
+    public static List<String> searchVivadoLog(List<String> log, String key) {
+        List<String> results = new ArrayList<>();
+        for(String l : log) {
+            if(l.contains(key)) {
+                results.add(l);
+            }
+        }
+        return results;
+    }
+
+    public static List<String> runVivadoTask(String runDir, String tclScript, boolean verbose) {
```

Comment:
Please add Javadoc for this method. I would also suggest naming it the more specific `runTcl{File,Command}`.

It is ambiguous without looking at the method body whether the `tclScript` parameter is a Tcl script file, or a Tcl command. Ideally, we would have both, with the latter building on the former by writing the command to a temp file and sourcing that.

Reply:
added Javadoc and overloaded this method to support single tcl commands or a script: https://github.com/Xilinx/RapidWright/pull/684/commits/23c2bd5ebab8579a71e576d8f85be5897fa2f463